pkg/blockchain/rpcLogStreamer.go

```diff
@@ -258,44 +264,80 @@ backfillLoop:
 
 func (r *RpcLogStreamer) GetNextPage(
 	ctx context.Context,
-	config ContractConfig,
-	fromBlock uint64,
-) (logs []types.Log, nextBlock *uint64, highestBlock uint64, err error) {
-	contractAddress := config.Address.Hex()
+	cfg ContractConfig,
+	fromBlockNumber uint64,
+	fromBlockHash []byte,
+) (logs []types.Log, nextBlockNumber *uint64, nextBlockHash []byte, highestBlock uint64, err error) {
+	contractAddress := cfg.Address.Hex()
+
+	if fromBlockNumber > 0 {
+		block, err := r.client.BlockByNumber(ctx, big.NewInt(int64(fromBlockNumber+1)))
```

Comment:
I am not sure how all the code connects. We return "current-block+1" as the next block down below. And here we also want to start at +1. I can't quite convince myself that there isn't some off-by-one situation here